lahja/asyncio/endpoint.py

```diff
@@ -111,25 +115,70 @@ def __init__(self, reader: StreamReader, writer: StreamWriter) -> None:
             raise RemoteDisconnected()
 
 
-class InboundConnection:
+class RemoteEndpoint:
     """
-    A local Endpoint might have several ``InboundConnection``s, each of them represents a remote
-    Endpoint which has connected to the given Endpoint and is attempting to send it messages.
+    Represents a connection to another endpoint.  Connections *can* be
+    bi-directional with messages flowing in either direction.
+
+    A 'message' can be any of:
+
+    - ``SubscriptionsUpdated``
+            broadcasting the subscriptions that the endpoint on the other side
+            of this connection is interested in.
+    - ``SubscriptionsAck``
+            acknowledgedment of a ``SubscriptionsUpdated``
+    - ``Broadcast``
+            an event meant to be processed by the endpoint.
     """
 
+    logger = logging.getLogger("lahja.endpoint.InboundConnection")
+
     def __init__(
-        self, conn: Connection, new_msg_func: Callable[[Broadcast], None]
+        self,
+        name: Optional[str],
```

Comment:
The `name` is an implicit part of the connection.  When a server receives an incoming connection over the socket it doesn't know the name of the endpoint on the other side of the connection.  This is the concept of a *half* connection.

I however now realize, that with subscriptions, we can potentially get rid of the distinction between half/full connections since we still have a mechanism for determining which endpoints are acceptable to send over.

However, I am going to open an issue to deal with this separately.

Reply:
https://github.com/ethereum/lahja/issues/96